ocaml/sdk-gen/component-test/jsonrpc-client/go/main_test.go

```diff
@@ -50,3 +59,650 @@ func TestLoginSuccess(t *testing.T) {
+		Order:                       0,
+		VGPUs:                       []xenapi.VGPURef{},
+		AttachedPCIs:                []xenapi.PCIRef{},
+		SuspendSR:                   "OpaqueRef:5ec2d621-7b62-f571-d38b-754341a973e5",
+		Version:                     0,
+		GenerationID:                "209202393574174067:5365773919542227445",
+		HardwarePlatformVersion:     2,
+		HasVendorDevice:             false,
+		RequiresReboot:              false,
+		ReferenceLabel:              "windows-11",
+		DomainType:                  "hvm",
+		NVRAM:                       map[string]string{"EFI-variables": "VkFSUwIAAAAsAAAAA"},
+		PendingGuidances:            []xenapi.UpdateGuidances{},
+		PendingGuidancesRecommended: []xenapi.UpdateGuidances{},
+		PendingGuidancesFull:        []xenapi.UpdateGuidances{},
+	}
+	var vmRecord1 = xenapi.VMRecord{
+		UUID: "060aee48-7ded-d2bd-6f0f-f224bd99c960",
+		AllowedOperations: []xenapi.VMOperations{
+			"changing_dynamic_range",
+			"migrate_send",
+			"pool_migrate",
+			"changing_VCPUs_live",
+			"suspend",
+			"hard_reboot",
+			"hard_shutdown",
+			"clean_reboot",
+			"clean_shutdown",
+			"pause",
+			"checkpoint",
+			"snapshot",
+		},
+		CurrentOperations: map[string]xenapi.VMOperations{},
+		NameLabel:         "ubuntu2204",
+		IsATemplate:       false,
+		SuspendVDI:        "OpaqueRef:NULL",
+		ResidentOn:        "OpaqueRef:4a0fde9c-5709-4f66-9968-64fb6162bd97",
+		MemoryStaticMax:   4294967296,
+		VBDs: []xenapi.VBDRef{
+			"OpaqueRef:4a54632c-a6b1-5535-4ef4-0d36d85e07df",
+			"OpaqueRef:56d316f7-e8d0-54f0-2492-a0952c6946c4",
+		},
+		HVMShadowMultiplier: 1.0,
+		Platform: map[string]string{
+			"acpi":         "1",
+			"apic":         "true",
+			"device-model": "qemu-upstream-compat",
+			"device_id":    "0001",
+			"hpet":         "true",
+			"nx":           "true",
+			"pae":          "true",
+			"timeoffset":   "0",
+			"vga":          "std",
+			"videoram":     "8",
+			"viridian":     "false",
+		},
+		Domid:        45,
+		SnapshotTime: time.Date(2023, time.July, 29, 13, 20, 01, 0, time.UTC),
+	}
+	var expectedResult = map[xenapi.VMRef]xenapi.VMRecord{
+		"OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13": vmRecord0,
+		"OpaqueRef:7cdfad16-7e0e-411f-1b6c-4308fd33b164": vmRecord1,
+	}
+	result, err := xenapi.VM.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Log("The result returned not the same with expected -> The XAPI outcome diverges from the anticipated value.")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMImport(t *testing.T) {
+	session, err := GetSession("test_id3")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var srRef xenapi.SRRef = "OpaqueRef:5ec2d621-7b62-f571-s38r-754341a973e5"
+	var VMImportURL = "http://vm_import_url"
+	var expectedResult = []xenapi.VMRef{"OpaqueRef:5ec2d621-7b62-f571-v38m-754341a973e5"}
```

Comment:
The expected result should be read from the spec. All tests should do this.

Reply:
I tried json.Unmarshal, but seems sometimes it could not parse the correct value from json file, after discussion, I will with @xueqingz  I reduce the field of xenapi.VMRecord, only keep one full records and then later we just keep the values that has different types or special values e.g. Inf or NaN.

Reply:
Hi @minglumlu , not able to simply implement read form json file and get map then covert to []xenapi.VMRef, even for a simple []xenapi.VMRef, thus also won't change based on this comment.